pmd-core/src/test/java/net/sourceforge/pmd/lang/rule/xpath/SaxonXPathRuleQueryTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.rule.xpath;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sourceforge.pmd.RuleContext;
+import net.sourceforge.pmd.lang.ast.DummyNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+public class SaxonXPathRuleQueryTest {
+
+    @Test
+    public void testListAttribute() {
+        DummyNodeWithListAndEnum dummy = new DummyNodeWithListAndEnum(1);
+
+        SaxonXPathRuleQuery query = createQuery("//dummyNode[@List = \"A\"]");
+        List<Node> result = query.evaluate(dummy, new RuleContext());
+        Assert.assertEquals(1, result.size());
+
+        query = createQuery("//dummyNode[@List = \"B\"]");
+        result = query.evaluate(dummy, new RuleContext());
+        Assert.assertEquals(1, result.size());
+
+        query = createQuery("//dummyNode[@List = \"C\"]");
```

Comment:
This looks like those three statements are duplicated and could be extracted to a single method call. Eg `assertResultSize(0, "//dummyNode[@List = \"C\"]");`

Reply:
Fixed.